File: ember-projects/linl/app/models/retirement.js
import DS from 'ember-data';

export default DS.Model.extend({
  user: DS.belongsTo('user'),

  name: DS.attr('string'),
  item_type: DS.attr('string'),
  monthly_cents: DS.attr('number'),
  total_amount_cents: DS.attr('number'),
  contributions_cents: DS.attr('number'),
  employer_contributions_cents: DS.attr('number'),
  start_date: DS.attr('date'),
  end_date: DS.attr('date'),
  user_owned: DS.attr('boolean')
});

File: rails-projects/linl-api/app/controllers/api/v1/entities/retirement.rb
module API
  module V1
    module Entities
      class RetirementEntity < Grape::Entity
        expose :id, :user_id, :name, :item_type, :monthly_cents, 
          :total_amount_cents, :contributions_cents, :employer_contributions_cents,
          :start_date, :end_date, :created_at, :updated_at
      end
    end
  end
end
